### PR TITLE
Change layer name logic

### PR DESCRIPTION
**Pull request by NihalHarish**
### Description of changes:
- Ensure that the name of the input and output layers do not change for different modes
- Add a test that verifies name strings with a regex pattern

#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/tensorflow2/test_keras.py [hunk 1]
```diff
@@ -29,6 +30,19 @@
...
+            # WA for TF issue https://github.com/tensorflow/tensorflow/issues/36279
+            tf.keras.layers.Flatten(input_shape=(28, 28, 1)),
+            tf.keras.layers.Dense(128, activation="relu"),
+            tf.keras.layers.Dropout(0.2),
+            tf.keras.layers.Dense(10, activation="softmax"),
+        ]
+    )
+    return model
+
+
```
Comment (NihalHarish) [inline tests/tensorflow2/test_keras.py:45]: refactored the model def to a fn

File: tests/tensorflow2/test_keras.py [hunk 2]
```diff
@@ -549,6 +545,28 @@ def test_include_regex(out_dir, tf_eager_mode):
...
 
 
+@pytest.mark.slow
+def test_layer_names(out_dir, tf_eager_mode):
+    hook = smd.KerasHook(
+        out_dir,
+        save_config=SaveConfig(save_interval=9),
+        include_collections=[CollectionKeys.LAYERS],
+    )
+    helper_keras_fit(
```
Comment (vandanavk) [inline tests/tensorflow2/test_keras.py:555]: add a gradtape test too?

File: smdebug/tensorflow/keras.py [hunk 3]
```diff
@@ -757,6 +757,13 @@ def _save_layer_values(self, logs):
         for layer_name, layer_input, layer_output in logs:
             # Cast layer_name to str since it can also be of type bytes
             # when run with mirrored strategy
+            if isinstance(layer_name, tf.Tensor):
```
Comment (vandanavk) [inline smdebug/tensorflow/keras.py:760]: isn't this problem there with other tensors that are saved from logs? gradients, metrics etc?

Comment (NihalHarish) [reply smdebug/tensorflow/keras.py:760]: Gradients already has this logic.
The problem happens from passing strings from the tf layer; so it will not happen with metrics.

**Review by vandanavk — APPROVED** [review]

File: tests/tensorflow2/test_keras.py [hunk 4]
```diff
@@ -549,6 +545,44 @@ def test_include_regex(out_dir, tf_eager_mode):
...
+    )
+
+    tr = create_trial_fast_refresh(out_dir)
+    tnames = tr.tensor_names(collection=CollectionKeys.LAYERS)
+    pattern = r"^(flatten|dense|dropout)(_\d+)?\/(inputs|outputs)"
+    for tname in tnames:
+        assert re.match(pattern=pattern, string=tname) is not None
+
+
+@pytest.mark.slow
```
Comment (vandanavk) [inline tests/tensorflow2/test_keras.py:570]: skip if non eager?

Comment (NihalHarish) [reply tests/tensorflow2/test_keras.py:570]: tf_eager is  not being passed as an argument here.
